utils/watchcat/files/watchcat.init

```diff
@@ -90,53 +90,42 @@ load_watchcat() {
 		logger -p user.err -t "watchcat" "reboot program $1 not started - $error"
 		return
 	}
+	
+	procd_open_instance
 
+	# Need to conditionally run mode functions because they have different signatures
 	case "$mode" in
 	periodic_reboot)
-		/usr/bin/watchcat.sh "periodic_reboot" "$period" "$forcedelay" &
-		logger -p user.info -t "watchcat" "started task (mode=$mode;period=$period;forcedelay=$forcedelay)"
+		procd_set_param command /usr/bin/watchcat.sh "periodic_reboot" "$period" "$forcedelay"
 		;;
 	ping_reboot)
-		/usr/bin/watchcat.sh "ping_reboot" "$period" "$forcedelay" "$pinghosts" "$pingperiod" "$pingsize" &
-		logger -p user.info -t "watchcat" "started task (mode=$mode;period=$period;pinghosts=$pinghosts;pingperiod=$pingperiod;forcedelay=$forcedelay;pingsize=$pingsize)"
+		procd_set_param command /usr/bin/watchcat.sh "ping_reboot" "$period" "$forcedelay" "$pinghosts" "$pingperiod" "$pingsize"
 		;;
 	restart_iface)
-		/usr/bin/watchcat.sh "restart_iface" "$period" "$pinghosts" "$pingperiod" "$pingsize" "$interface" "$mmifacename" &
-		logger -p user.info -t "watchcat" "started task (mode=$mode;period=$period;pinghosts=$pinghosts;pingperiod=$pingperiod;pingsize=$pingsize;interface=$interface;mmifacename=$mmifacename;unlockbands=$unlockbands)"
+		procd_set_param command /usr/bin/watchcat.sh "restart_iface" "$period" "$pinghosts" "$pingperiod" "$pingsize" "$interface" "$mmifacename"
 		;;
 	*)
 		echo "Error starting Watchcat service. Invalid mode selection: $mode"
 		;;
 	esac
 
-	echo $! >>"${PIDFILE}.pids"
+	procd_set_param respawn ${respawn_threshold:-3600} ${respawn_timeout:-5} ${respawn_retry:-5}
+	procd_close_instance
 }
 
-stop() {
-	if [ -f "${PIDFILE}.pids" ]; then
-		logger -p user.info -t "watchcat" "stopping all tasks"
-
-		while read pid; do
-			kill -KILL "$pid"
-		done <"${PIDFILE}.pids"
-
-		rm "${PIDFILE}.pids"
-
-		logger -p user.info -t "watchcat" "all tasks stopped"
-	else
-		logger -p user.info -t "watchcat" "no tasks running"
-	fi
+start_service() {
+	config_load watchcat
+	config_foreach config_watchcat watchcat
+	procd_set_param file /etc/config/watchcat
 }
 
-start() {
-	[ -f "${PIDFILE}.pids" ] && stop
-
-	config_load system
-	if [ -n "$(uci show system.@watchcat[0])" ]; then # at least one watchcat section exists
-		logger -p user.info -t "watchcat" "starting all tasks"
-		config_foreach load_watchcat watchcat
-		logger -p user.info -t "watchcat" "all tasks started"
-	else
-		logger -p user.info -t "watchcat" "no tasks defined"
-	fi
+service_triggers()
+{
+	procd_add_reload_trigger "watchcat"
 }
+
+reload_service()
+{
+	stop
+	start
+}
```

Comment:
No newline?

Reply:
I can add it, but I cannot find a consistent pattern of examples or style guide to follow.  For instance, this one does not have a newline. https://github.com/openwrt/packages/blob/master/mail/emailrelay/files/emailrelay.init

Maybe I can start a style guide page on the wiki?

Reply:
That would not be bad if you can do that.

Reply:
I've started it here: https://openwrt.org/code_style_guide. Let's continue discussion on the mailing list, wiki and forum.

Reply:
I have added the new line at the end, as that should be the standard.